src/WTG.Analyzers.Test/TestData/QueryLinqAnalyzer/Enumerable/Item/Result.cs

```diff
@@ -0,0 +1,16 @@
+using System.Linq;
+
+static class Foo
+{
+	public static Item First1(IQueryable<Item> items) => items.AsEnumerable().First(Filter);
+	public static Item Single1(IQueryable<Item> items) => items.AsEnumerable().Single(Filter);
+	public static Item FirstOrDefault1(IQueryable<Item> items) => items.AsEnumerable().FirstOrDefault(Filter);
+	public static Item SingleOrDefault1(IQueryable<Item> items) => items.AsEnumerable().SingleOrDefault(Filter);
```

Comment:
The ideal fix in these examples would have been to move the filter body from the invoked method to the query expression so that the filtering can be performed "on the server":
```csharp
public static Item First1(IQueryable<Item> items) => items.First(x => x.Name.StartsWith("Foo"));
public static Item Single1(IQueryable<Item> items) => items.Single(x => x.Name.StartsWith("Foo"));
public static Item FirstOrDefault1(IQueryable<Item> items) => items.FirstOrDefault(x => x.Name.StartsWith("Foo"));
public static Item SingleOrDefault1(IQueryable<Item> items) => items.SingleOrDefault(x => x.Name.StartsWith("Foo"));
```

But actually trying to achieve that would be problematic. Adding a code fix that inserts `.AsEnumerable()` would encourage people to use that as a crutch to make the error go away rather than fix the underlying issue.

Reply:
I've previously caused a production problem by accidentally selecting >1M records from a database in a similar manner to what the `.AsEnumerable()` code fix suggests, I would be very hesitant for the IDE to _suggest_ it to people in any context.